pkg/types/gateway/ratelimiter.go

```diff
@@ -0,0 +1,69 @@
+package gateway
```

Comment:
nit (after seeing the core PR): maybe we can put it in some "lib"/"shared"/"utils" package instead?

Reply:
makes sense. can definitely be used outside of gateway. how about a new top-level package? `pkg/ratelimit`